src/internal/DotStarEsp32DmaSpiMethod.h

```diff
@@ -0,0 +1,335 @@
+
+        _ssPin = ss;
+
+        esp_err_t ret;
+        spi_bus_config_t buscfg;
+        memset(&buscfg, 0x00, sizeof(buscfg));
+
+        buscfg.miso_io_num = dat1;
+        buscfg.mosi_io_num = dat0;
+        buscfg.sclk_io_num = sck;
+        buscfg.quadwp_io_num = dat2;
+        buscfg.quadhd_io_num = dat3;
+        buscfg.max_transfer_sz = _spiBufferSize;
+
+        //Initialize the SPI bus
+        ret=spi_bus_initialize(T_SPIBUS::SpiHostDevice, &buscfg, T_SPIBUS::dmaChannel);
+        ESP_ERROR_CHECK(ret);
+
+        InitSpiDevice();
+    }
+
+    void Initialize(int8_t sck, int8_t miso, int8_t mosi, int8_t ss)
+    {
+        Initialize(sck, mosi, miso, -1, -1, ss);
+    }
+
+    // If pins aren't specified, initialize bus with just the default SCK and MOSI pins for the SPI peripheral (no SS, no >1-bit pins)
+    void Initialize()
+    {
+        if (T_SPIBUS::SpiHostDevice == VSPI_HOST)
+        {
+            Initialize(SCK, -1, MOSI, -1, -1, -1);
+        }
+        else
+        {
+            Initialize(14, -1, 13, -1, -1, -1);
+        }
+    }
+
+    void Update(bool)
+    {
+        while(!IsReadyToUpdate());
+
+        memcpy(_dmadata, _data, _spiBufferSize);
+
+        memset(&_spiTransaction, 0, sizeof(spi_transaction_t));
+        _spiTransaction.length = (_spiBufferSize) * 8; // in bits not bytes!
+        if (T_SPIBUS::parallelBits == 1)
+        {
+            _spiTransaction.flags = 0;
+        }
+        if (T_SPIBUS::parallelBits == 2)
+        {
+            _spiTransaction.flags = SPI_TRANS_MODE_DIO;
+        }
+        if (T_SPIBUS::parallelBits == 4)
+        {
+            _spiTransaction.flags = SPI_TRANS_MODE_QIO;
+        }
+        _spiTransaction.tx_buffer = _dmadata;
+
+        esp_err_t ret = spi_device_queue_trans(_spiHandle, &_spiTransaction, 0);  //Transmit!
+        assert(ret == ESP_OK);            //Should have had no issues.
+    }
+
+    uint8_t* getData() const
+    {
+        return _data + _sizeStartFrame;
+    };
+
+    size_t getDataSize() const
+    {
+        return _sizePixelData;
+    };
+
+    void applySettings(const SettingsObject& settings)
+    {
+        _speed.applySettings(settings);
+        if (_spiHandle)
+        {
+            DeInitSpiDevice();
+            InitSpiDevice();
+        }
+    }
+
+private:
+    void InitSpiDevice()
+    {
+        spi_device_interface_config_t devcfg = {};
+
+        devcfg.clock_speed_hz = _speed.Clock;
+        devcfg.mode = 0;                 //SPI mode 0
+        devcfg.spics_io_num = _ssPin;    //CS pin
+        devcfg.queue_size = 1;
+        if (T_SPIBUS::parallelBits == 1)
+        {
+            devcfg.flags=0;
+        }
+        if (T_SPIBUS::parallelBits >= 2)
+        {
+            devcfg.flags=SPI_DEVICE_HALFDUPLEX;
+        }
+
+        //Allocate the LEDs on the SPI bus
+        esp_err_t ret=spi_bus_add_device(T_SPIBUS::SpiHostDevice, &devcfg, &_spiHandle);
```

Comment:
please review all your locations of operators without spaces around them and fix them.
`ret=spiblah` should be `ret = spiblah`